pkg/channel/distributed/common/kafka/sarama/sarama.go

```diff
@@ -296,21 +301,29 @@ func LoadSettings(ctx context.Context) (*sarama.Config, *commonconfig.EventingKa
 		return nil, nil, err
 	}
 
+	eventingKafkaConfig, err := LoadEventingKafkaSettings(configMap)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// Merge The Sarama Settings In The ConfigMap Into A New Base Sarama Config
+	saramaConfig, err := MergeSaramaSettings(nil, configMap)
+
+	return saramaConfig, eventingKafkaConfig, err
+}
+
+func LoadEventingKafkaSettings(configMap *corev1.ConfigMap) (*commonconfig.EventingKafkaConfig, error) {
 	// Validate The ConfigMap Data
 	if configMap.Data == nil {
-		return nil, nil, fmt.Errorf("attempted to load configuration from empty configmap")
+		return nil, fmt.Errorf("attempted to load configuration from empty configmap")
 	}
```

Comment:
check `configMap` is nill as well (before configMap.Data check) ?

Reply:
good idea, will do